CoffeeMachine.java

```diff
@@ -0,0 +1,150 @@
+import jdk.swing.interop.SwingInterOpUtils;
+
+import java.util.Scanner;
+
+public class CoffeeMachine {
+Scanner object=new Scanner(System.in);
+private int espressoSingleShot=0;
+private int DoubleShotDoppio=0;
+private int Lungo=0;
+private int Macchiato=0;
+private int Cappuccino=0;
+private int coldWater;
+private int coffeeGrounds;
+
+    public void EspressoSingleShot() throws InterruptedException {
+        System.out.println("Would you like to add cold water(y/n)?");
+        char answer=object.next().charAt(0);
+        if(addWater(answer)){
+            this.coldWater=5; //golden ratio (zbog toga je hardcodirano)
+        System.out.println("Added "+ this.coldWater+ " ounces of cold water");
+        }
+            System.out.print("Put coffeeGrounds:");
+            this.coffeeGrounds = object.nextInt();
+            if (makeEspressoSingleShot(this.coffeeGrounds)) {
+                System.out.println("Your coffee is done");
+                this.espressoSingleShot++;
+                System.out.println("You made "+ this.espressoSingleShot+ " single shots so far!.");
+            } else {
+                System.out.println("Not the right amount of coffee grounds");
+            }
+    }
+private boolean makeEspressoSingleShot(int coffeeGrounds) throws InterruptedException {
+if(coffeeGrounds>=6 && coffeeGrounds<=8){
+    System.out.print("Brewing time(WAIT):");
+    Thread.sleep(20000);
+    return true;
+}
+else return false;
+}
+public void DoubleShotDoppio() throws InterruptedException {
+    System.out.println("Would you like to add cold water(y/n)?");
+    char answer=object.next().charAt(0);
+    if(addWater(answer)){
+        this.coldWater=5;
+        System.out.println("Added "+ this.coldWater+ " ounces of cold water");
+    }
+    System.out.print("Put coffeeGrounds:");
+    this.coffeeGrounds = object.nextInt();
+    if (makeDoubleShotDoppio(this.coffeeGrounds)) {
+        System.out.println("Your coffee is done");
+        this.DoubleShotDoppio++;
+        System.out.println("You made "+ this.DoubleShotDoppio+ " double shots so far!");
+    } else {
+        System.out.println("Not the right amount of coffee grounds");
+    }
+}
+    private boolean makeDoubleShotDoppio(int coffeeGrounds) throws InterruptedException {
+        if(coffeeGrounds>=18 && coffeeGrounds<=21){
+            System.out.print("Brewing time(WAIT):");
+            Thread.sleep(30000);

```

Comment:
This code will execute for a long time :) I understand you wanted to have a simulation but, for now, it is ok that you skip sleeping mode. This way, you will not need throws InterruptedException in method signatures